File: services/go/graphql/src/domain/entities/product.go
package entities

import "time"

type Product struct {
	ID            int        `json:"id"`
	Name          string     `json:"name"`
	Description   string     `json:"description"`
	Category      string     `json:"category"`
	Images        []string   `json:"images"`
	Price         float64    `json:"price"`
	StockQuantity int        `json:"stockQuantity"`
	CreatedAt     time.Time  `json:"createdAt"`
	UpdatedAt     *time.Time `json:"updatedAt"`
}

type CreateInput struct {
	Name          string
	Description   string
	Category      string
	Images        []string
	Price         float64
	StockQuantity int
}

type UpdateInput struct {
	Name          string
	Description   string
	Category      string
	Images        []string
	Price         float64
	StockQuantity int
}

File: services/rust/grpc/src/domain/entities/product.rs
use chrono::{DateTime, Utc};
use rust_decimal::Decimal;

#[derive(Debug, Clone, sqlx::FromRow)]
pub struct Product {
    #[sqlx(rename = "Id")]
    pub id: i32,
    #[sqlx(rename = "Name")]
    pub name: String,
    #[sqlx(rename = "Description")]
    pub description: String,
    #[sqlx(rename = "Category")]
    pub category: String,
    #[sqlx(rename = "Images")]
    pub images: Vec<String>,
    #[sqlx(rename = "Price")]
    pub price: Decimal,
    #[sqlx(rename = "StockQuantity")]
    pub stock_quantity: i32,
    #[sqlx(rename = "CreatedAt")]
    pub created_at: DateTime<Utc>,
    #[sqlx(rename = "UpdatedAt")]
    pub updated_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone)]
pub struct CreateProductInput {
    pub name: String,
    pub description: String,
    pub category: String,
    pub images: Vec<String>,
    pub price: Decimal,
    pub stock_quantity: i32,
}

#[derive(Debug, Clone)]
pub struct UpdateProductInput {
    pub name: String,
    pub description: String,
    pub category: String,
    pub images: Vec<String>,
    pub price: Decimal,
    pub stock_quantity: i32,
}
